src/shared/sam/cli/samCliErrors.ts

```diff
@@ -0,0 +1,81 @@
+/*!
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * Represents the original message as well as
+ * a shortened human-readable version, if possible.
+ */
+export interface Message {
+    original: string
+    short?: string
+}
+export type Messages = Message[]
+
+/**
+ * A function that can attempt to give a more concise, 'shortened',
+ * version of a message. Otherwise returning void if it is unable.
+ */
+export type MessageShortener = (message: string) => string | undefined
+export type MessageShortners = MessageShortener[]
+
+/**
+ * Attempt to give a more concise, 'shortened',
+ * version of each message.
+ * @param messages All the messages you want to try and shorten
+ * @param shorteners All the shortener functions to apply to each message
+ * @returns A list of message objects with a shortened message, if it exists.
+ */
+export function tryShortenMessages(messages: string[], shorteners: MessageShortners): Messages {
+    const result: Messages = []
+    for (const message of messages) {
+        const shortened = tryShortenMessage(message, shorteners)
+        if (shortened != undefined) {
+            result.push({ original: message, short: shortened })
+        } else {
+            result.push({ original: message })
+        }
+    }
+    return result
+}
+
+/**
+ * Attempt to give a more concise, 'shortened',
+ * version of a given message.
+ * @param message The message we want to try to shorten
+ * @param shorteners All the shorteners to apply to the message
+ * @returns The first succesfully shortened message
+ */
+export function tryShortenMessage(message: string, shorteners: MessageShortners): string | undefined {
+    for (const shorten of shorteners) {
+        const shortenedMessage = shorten(message)
+        if (shortenedMessage != undefined) {
+            return shortenedMessage
+        }
+    }
+}
+
+/**
+ * Shortens the missing python runtime error message from SAM Cli
+ */
+export function missingPythonRuntimeShortener(message: string): string | undefined {
+    const pattern = /Do you have python for runtime: (python\S+) on your PATH\??/
```

Comment:
If sam cli has localized messages, this won't work. That's why something more generic like looking for `Error:` would be preferrable.

Reply:
I'm just wondering how we are supposed to tackle task #2 (providing a more user friendly message) if we don't search for this specific message in IDE-8580.

Specifically for this case, the full error message is `Error: PythonPipBuilder:Validation - Binary validation failed for python, searched for python in following locations  : ['/usr/bin/python3'] which did not satisfy constraints for runtime: python3.7. Do you have python for runtime: python3.7 on your PATH?`

Reply:
> wondering how we are supposed to tackle task 2 of the SIM (providing a more user friendly message) if we don't search for this specific message

Good question! Let's change the SIM :)